src/core/reaction_ensemble.cpp

```diff
@@ -0,0 +1,2061 @@
+//method according to smith94x for the reaction ensemble at constant volume and temperature, for the reaction ensemble at constant pressure additionally employ a barostat!
```

Comment:
Unfortunately running clang-format makes the code unreadable since the variable/function names are long. Still I would not shorten the variable names since they are fully descriptive.